lib/unleash/configuration.rb

```diff
@@ -118,6 +113,14 @@ def merge(opts)
       self
     end
 
+    def generate_custom_http_headers
+      if self.custom_http_headers.respond_to?(:call)
+        self.custom_http_headers.call
+      else
+        self.custom_http_headers
+      end
```

Comment:
Would suggest this alternative syntax to make it more ruby idiomatic and easier to read:
```suggestion
      return self.custom_http_headers.call if self.custom_http_headers.respond_to?(:call)

      self.custom_http_headers.dup
```

Note that we duplicate the value of the `custom_http_headers` Hash, to ensure that they are not unexpectedly changed later on after the initialization of the client.

Reply:
The `Configuration#http_headers` method already creates two new instances of a Hash so it's not needed to duplicate. And if we add the duplication here it's just the Hash that gets passed to the merge function that is duplicated. Not the result of the `Configuration#https_headers` method.

Reply:
We would suggest changing the `merge` to be `merge!` in `Configuration#http_headers` to avoid creating two Hashes there and just merge it into the hash that is created in the beginning of the method.

Reply:
I agree that no duplication is needed in this case.

Feel free to use `merge!`, that does look more elegant.